src/core.ts

```diff
@@ -103,6 +105,8 @@ export const $ = new Proxy<Shell & Options>(
       let s
       if (Array.isArray(args[i])) {
         s = args[i].map((x: any) => $.quote(substitute(x))).join(' ')
+      } else if (is$Object(args[i])) {
+        s = dargs(args[i]).map((x: any) => $.quote(substitute(x))).join(' ')
```

Comment:
This doesn't support configuring the options for dargs.

Some shell commands may not match the syntax dargs uses by default, so they may want to [set options like `allowCamelCase`](https://github.com/sindresorhus/dargs#api). Also dargs doesn't seem to support generating the `foo -bar=baz` syntax some shell commands use (but perhaps dargs would consider adding this as an option?)

Reply:
@bburky I'll look more on foo -bar=baz as while coding this I've considered the requirements that were put on issue. If it's a valid possible use case I may think of adding this to dargs if not there.